oop2-encapsulate-inherit.js

```diff
@@ -1,41 +1,270 @@
-    
-    - For each class, add 2 properties and 2 methods to each.  The properties and methods should make sense, considering what the classes are supposed to be representing.
+}
+class Review {
+    #review;
+    #user;
+    constructor() {
+        this.#review = null;
+        this.#user = null;
+    }
+    getReview() {
+        return this.#review;
+    }
+    setReview(review) {
+        this.#review = review;
+    }
+    getUser() {
+        return this.#user;
+    }
+    setUser(user) {
+        this.#user = user;
+    }
 
+}
 
 
-    //your code here...
 
 
 
 
-/****************************************************************************************************************************************************************************************   
+// 2. Create the following classes:
+//     - Umbrella class.  It should have a an organization name.  It should only have a single instance.  Umbrella is the "Parent Organization" e.g. Allegis Group
+//     - Company class. It should have a company name and # of employees.  
+//       Create at least 3 different companies that are children of the Umbrella Organization.  (e.g. TEKsystems, Aerotek, etc.)
+//     - Site class. It should have the name of the company, and the location of the site.  It is a child class that inherits from the Company class.  Create 3 sites for each company.
+//     - Employee class. It should have a employee name, job title, and salary properties. 
+//      Create 10 employee instances: 1 CEO, 1 Manager, 1 Secretary, 2 Engineers, 1 Financial Officer, 1 Janitor, 1 Marketer, 1 HR Personnel, 1 Lawyer.  Feel free to add other employee roles at your own discretion.
+
+//     - For each class, add 2 properties and 2 methods to each.  The properties and methods should make sense, considering what the classes are supposed to be representing.
+
+
+
+//     //your code here...
+
+class Umbrella {
+    #name;
+    #totalCompanies
+    constructor() {
+
+        this.#name = "Allegis Group";
+        this.#totalCompanies = 3;
+
+    }
+    getUmbrellaName() {
+        return this.#name;
+    }
+    getTotalCompanies() {
+        return this.#totalCompanies;
+    }
+
+}
+class Companies extends Umbrella {
+    #companyName;
+    #numberofEmployees;
+    constructor(companyName, numberofEmployees) {
+        super();
+        this.#companyName = companyName;
+        this.#numberofEmployees = numberofEmployees;
+
+    }
+    getcompanyName() {
+        return this.#companyName;
+    }
+    getnumberofEmployees() {
+        return this.#numberofEmployees;
+    }
+    addEmployeeToRoster() {
+        this.#numberofEmployees++;
+    }
+    sayCompanyDetails() {
+        return `At ${this.#companyName}, we have ${this.#numberofEmployees} employees.`;
+    }
+
+}
+class Site extends Companies {
+    #siteCompanyName;
+    #siteLocation;
+    constructor(siteCompanyName, siteLocation) {
+        super();
```

Comment:
If Site extends Companies, it needs to take in all the values that companies and it's parents need in their constructors, then pass them on to `super()`